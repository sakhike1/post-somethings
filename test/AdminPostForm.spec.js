import { mount } from "@vue/test-utils";
import AdminPostForm from "@/components/Admin/AdminPostForm.vue";

describe("AdminPostForm", () => {
  it("emits submit event with edited post data", async () => {
    const wrapper = mount(AdminPostForm);

    // Fill in form fields
    await wrapper.setData({
      editedPost: {
        author: "John Doe",
        title: "Test Post",
        thumbnail: "thumbnail-url",
        content: "This is the content",
        previewText: "Preview text",
      },
    });

    // Simulate form submission
    await wrapper.find("form").trigger("submit.prevent");

    // Check if submit event is emitted with correct data
    expect(wrapper.emitted().submit[0][0]).toEqual({
      author: "John Doe",
      title: "Test Post",
      thumbnail: "thumbnail-url",
      content: "This is the content",
      previewText: "Preview text",
    });
  });

  it("navigates back on cancel", async () => {
    const $router = { push: jest.fn() };
    const wrapper = mount(AdminPostForm, {
      mocks: {
        $router,
      },
    });

    // Mock the $router.push method
    wrapper.vm.$router.push = $router.push;

    await wrapper.vm.onCancel();
    expect($router.push).toHaveBeenCalledWith("/admin");
    expect().toMatchSnapshot();
  });

  // Add more test cases as needed
});
