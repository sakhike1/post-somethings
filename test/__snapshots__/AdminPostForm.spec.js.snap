// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminPostForm navigates back on cancel 1`] = `undefined`;
